common/app/common/TrailsToShowcase.scala

```diff
@@ -318,15 +319,54 @@ object TrailsToShowcase {
   private def webUrl(content: PressedContent): Option[String] =
     content.properties.maybeContent.map(_.metadata.webUrl)
 
-  def titleOfLengthFrom(length: Int, content: PressedContent): Either[Seq[String], String] = {
+  private def titleOfLengthFrom(length: Int, content: PressedContent): Either[Seq[String], String] = {
+    val (_, title) = inferPanelTitleAndTitleFrom(content)
+    Right(title)
+      .filterOrElse(_.nonEmpty, Seq("Heading is empty"))
+      .filterOrElse(_.length <= length, Seq(s"The headline '$title' is longer than " + length + " characters"))
+  }
+
+  private def panelTitleFrom(content: PressedContent): Either[Seq[String], Option[String]] = {
+    val (maybePanelTitle, _) = inferPanelTitleAndTitleFrom(content)
+    maybePanelTitle
+      .map { panelTitle =>
+        Right(Some(panelTitle))
+          .filterOrElse(_ => panelTitle.nonEmpty, Seq(s"Panel title in headline '${content.header.headline}' is empty"))
+          .filterOrElse(
+            _ => panelTitle.length <= MaxLengthForPanelTitle,
+            Seq(s"The panel title '$panelTitle' is longer than " + MaxLengthForPanelTitle + " characters"),
+          )
+      }
+      .getOrElse {
+        Right(None)
+      }
+  }
+
+  private def mandatoryPanelTitleFrom(content: PressedContent): Either[Seq[String], String] = {
+    val (maybePanelTitle, _) = inferPanelTitleAndTitleFrom(content)
+    maybePanelTitle
+      .map { panelTitle =>
+        Right(panelTitle)
+          .filterOrElse(_ => panelTitle.nonEmpty, Seq(s"Panel title in headline '${content.header.headline}' is empty"))
+          .filterOrElse(
+            _ => panelTitle.length <= MaxLengthForPanelTitle,
+            Seq(s"The panel title '$panelTitle' is longer than " + MaxLengthForPanelTitle + " characters"),
+          )
+      }
+      .getOrElse {
+        Left(Seq(s"Could not find a panel title in the first trail headline '${content.header.headline}'"))
+      }
+  }
+
+  private def inferPanelTitleAndTitleFrom(content: PressedContent): (Option[String], String) = {
     val trailTitle = TrailsToRss.stripInvalidXMLCharacters(content.header.headline)
-    Some(trailTitle)
-      .filter(_.nonEmpty)
-      .filter(_.length <= length)
-      .map(Right(_))
-      .getOrElse(
-        Left(Seq(s"The headline '$trailTitle' is longer than " + length + " characters")),
-      )
+    // Look for panel title delimiter
+    val pipeDelimited = trailTitle.split(PanelTitleInHeadlineDelimiter).toSeq
+    val (maybePanelTitle, title) = pipeDelimited.length match {
```

Comment:
No particular benefit in assigning these to (maybePanelTitle, title), and then returning the tuple.
Also, `if/else` would be clearer than a pattern match here.